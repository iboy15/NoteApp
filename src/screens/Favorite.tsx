import {
  ActivityIndicator,
  FlatList,
  StyleSheet,
} from 'react-native'
import {  View } from '../components/Themed'
import {  RootTabScreenProps } from '../../types'
import { getColor, width } from '../utils'
import useColorScheme from '../hooks/useColorScheme'
import {useMain} from '../hooks/useMain'
import NoteCard from '../components/NoteCard'
import { useFocusEffect } from '@react-navigation/native'
import { useCallback, useEffect } from 'react'

export default function Favorite({ navigation }: RootTabScreenProps<'Notes'>) {
  const colorScheme = useColorScheme()
  const { getData, notesData, loading, setAsFavorite,needRefreshFavorite,setIsNeedRefreshFavorite } = useMain()

  useFocusEffect(
    useCallback(() => {
    console.log('needRefreshFavorite', needRefreshFavorite)
      if(needRefreshFavorite){
        getData()
        setIsNeedRefreshFavorite(false)
      }
    }, [needRefreshFavorite])
  ) 

  const filteredData = notesData?.length !=0 ?
  notesData.filter(item => !item.is_archived && item.is_favorite) :[]


  return (
    <View style={styles.container}>
      {loading ? (
        <ActivityIndicator size='large' />
      ) : (
        <FlatList
          contentContainerStyle={{alignItems:'flex-start'}}
          style={{ width: '95%', }}
          onRefresh={getData}
          refreshing={loading}
          keyExtractor={(i, index) => index.toString()}
          numColumns={2}
          horizontal={false}
          data={filteredData}
          renderItem={({ item, index }) => {
            const color = getColor(colorScheme)
              return (
                <NoteCard
                  onPress={() =>
                    navigation.navigate('ModalDetail', {
                      ...item,
                      color,
                      index,
                    })
                  }
                  title={item.title}
                  body={item.body}
                  created_at={item.created_at}
                  is_favorite={item.is_favorite}
                  color={color}
                  setAsFavorite={()=>setAsFavorite(index)}
                />
              )
          }}
        />
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    paddingTop: 20,
  },
  title: { fontWeight: 'bold', marginBottom: 5 },
  listItem: {
    width: width / 2 - 20,
    margin: 10,
    borderRadius: 10,
    overflow: 'hidden',
    padding: 10,
  },
  bottomContainer: {
    flexDirection: 'row',
    backgroundColor: 'transparent',
    marginTop: 'auto',
  },
  favoriteBtn: {
    height: 28,
    width: 28,
    borderRadius: 19,

    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 'auto',
    paddingLeft: 1,
  },
  body: { marginBottom: 10 },
})
