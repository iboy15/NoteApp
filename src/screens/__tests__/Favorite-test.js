import * as React from 'react';
import renderer from 'react-test-renderer';
import Favorite from '../Favorite'
import MockedNavigator from '../../navigation/MockedNavigator'

it(`renders correctly`, async () => {
  const tree = await renderer.create(<MockedNavigator  component={Favorite} />).toJSON();

  expect(tree).toMatchSnapshot();
});
