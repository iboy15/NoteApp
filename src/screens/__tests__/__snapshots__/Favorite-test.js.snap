// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="MockedScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              Object {
                "alignItems": "center",
                "flex": 1,
                "paddingTop": 20,
              },
            ]
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "alignItems": "flex-start",
              }
            }
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onRefresh={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            refreshControl={
              <RefreshControlMock
                onRefresh={[Function]}
                refreshing={false}
              />
            }
            refreshing={false}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "width": "95%",
              }
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <RCTRefreshControl />
            <View />
          </RCTScrollView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
